core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -239,10 +258,11 @@ private void contractNodes() {
         // no witness path can be found. this is not really what we want, but changing it requires re-optimizing the
         // graph contraction parameters, because it affects the node contraction order.
         meanDegree = prepareGraph.getAllEdges().length() / prepareGraph.getNodes();
+        initSize = sortedNodes.getSize();
+        // todo: why do we start counting levels with 1 ??
```

Comment:
I think I was using it to distinguish uninitialized nodes from nodes with level=0 but now that we initialize all nodes with maxLevel at start this is probably no longer necessary.

Reply:
Ok I would suggest starting the count at 0 then, this would have saved me some nerves the first time I did not realize it starts at 1 :)

Reply:
Oh, ok. :+1: